substrate/frame/staking/src/ledger.rs

```diff
@@ -171,7 +172,12 @@ impl<T: Config> StakingLedger<T> {
 			return Err(Error::<T>::NotStash)
 		}
 
-		T::Currency::set_lock(STAKING_ID, &self.stash, self.total, WithdrawReasons::all());
+		if !<Delegatees<T>>::contains_key(&self.stash) {
+			// if the stash is a delegatee, funds are only delegated and already locked/held in the
```

Comment:
How about we lock the delegatee funds here too? That's one of the goals of having the `StakingLedger` struct wrapper., to consolidate all the ledger and lock logic under the same struct.

Reply:
Or maybe my comment doesn't make much sense in this context. The way that funds in the ledger of delegatees are accounted for is different from other ledgers, since they are accounted based on the locked and delegated stake from delegators. In that case, we should just skit the set lock here. let's discuss about this.

Reply:
Yeah I was thinking about how to abstract this. We could have a LockProvider trait and StakingLedger generic over this. The LockProviders would be either currency or staking delegate pallet. I will work on making this cleaner.